### PR TITLE
Employees Working Hour Report

### PR DESCRIPTION
**Pull request by varungadde99**
- Sorting on Name + Date
 - All dates with working hours > 9 are fetched.
 - All dates of a particular user are arranged followed by next user.

### CODE REVIEW

**Comment by varungadde99** [comment]
[Employees Working Hour Report](https://trello.com/c/64t4AD2g/177-employees-working-hour-report)

File: app/models/time_sheet.rb [hunk 1]
```diff
@@ -561,19 +561,22 @@ def self.generate_and_send_weekend_report(holiday_list, start_date)
...
           timesheets.keys.each do |key|
             total_duration = timesheets[key].map(&:duration).sum
             reports << [  timesheets[key].first.date.to_s,
-                          timesheets[key].first.user.name,
-                          timesheets[key].first.project.name,
-                          formatted_duration(total_duration)]
+                        timesheets[key].first.user.name,
+                        timesheets[key].first.project.name,
+                        formatted_duration(total_duration)]
           end
```
Comment (Sameer-Tilak) [inline app/models/time_sheet.rb:580]: Though the code looks correct in functionality, it is not optimised. Code is iterating on users and within each use each date. This is wrong way of coding. You can deploy this code, but add new ticket to optimise this code. this. Think on the logic, discuss it with me before development.

**Comment by Sameer-Tilak** [comment]
Looks ok for now.  can be deployed. But see the comment above and work on that.

File: app/models/time_sheet.rb (hunk 1 above)
Comment (varungadde99) [reply app/models/time_sheet.rb:580]: Ok sir. Thanks for the feedback. Will work on the same to optimize the code.

**Pull request closed without merging**